runtime/nvqir/NVQIR.cpp

```diff
@@ -207,16 +207,16 @@ void __quantum__rt__qubit_release_array(Array *arr) {
 Qubit *__quantum__rt__qubit_allocate() {
   ScopedTraceWithContext("NVQIR::allocate_qubit");
   __quantum__rt__initialize(0, nullptr);
-  auto qubitIdx = nvqir::getCircuitSimulatorInternal()->allocateQubit();
+  auto qubitIdx = nvqir::getCircuitSimulatorInternal()->allocateQudit();
   auto qubit = std::make_unique<Qubit>(qubitIdx);
   nvqir::allocatedSingleQubits.emplace_back(std::move(qubit));
   return nvqir::allocatedSingleQubits.back().get();
 }
 
 /// @brief Once done, release that qubit
 void __quantum__rt__qubit_release(Qubit *q) {
-  ScopedTraceWithContext("NVQIR::release_qubit");
-  nvqir::getCircuitSimulatorInternal()->deallocate(q->idx);
+  cudaq::ScopedTrace trace("NVQIR::release_qubit");
```

Comment:
I'd suggest using `ScopedTraceWithContext` as that will show the source file and line number.
```suggestion
  ScopedTraceWithContext("NVQIR::release_qubit");
```